### ISSUE TITLE
Udev rules for dev environments

### ISSUE THREAD

**Issue by blaggacao** [numtide/devshell#51]
For some [dev envronments](https://platformio.org/), there is a use case to install [udev rules](https://raw.githubusercontent.com/platformio/platformio/develop/scripts/99-platformio-udev.rules) for non-root device access (`MODE="0666"`).

The current nixos-specific workarround is to use `services.udev.package = [ pkgs.platformio ];`.

However, this does not properly encapsulate the desired devshell.

Maybe this falls into the same problem category as managing PKI or DNS for development environments.

Is there an "encapsulatable devshell approach" to this problem (existing and feasible)?

Further links:

- https://wiki.archlinux.org/index.php/udev

**Comment by zimbatm** [comment]
yeah, that's rather unfortunate. I think this fits into "we need a system configuration change tool" category, which is a separate but complementary project.

**Comment by blaggacao** [comment]
Do you want me to close? Or may I suggest a `delegate-to-complementary-tooling` flag?

**Comment by 06kellyjac** [comment]
what would `delegate-to-complementary-tooling` flag achieve that a script in `shellHook` (not sure what the equivalent is in `devshell`) or setting up a script in `commands` can't do?

**Comment by blaggacao** [comment]
My bad. Wording has been deficient. Updated my previous post. Thanks for making me realize.

**Comment by zimbatm** [comment]
Let's keep those around, it doesn't hurt to have the periphery of the project also be documented through issues like that.
